### DISCUSSION TITLE
MusicBrainz plugin separation/development

### DISCUSSION THREAD

**Discussion by lyarenei** [jellyfin/jellyfin#10718] [General]
Hi, I'm currently looking at adding stuff to the MusicBrainz plugin and when I started working on that, I noticed there are two implementations of the plugin. One is directly in the JF server and the other one is in its own repo. Now, my assumption is that the long term plan is to have the plugin fully separated from the server.

Is there some sort of plan/checklist what needs to happen in order to finish the split? I can think of at least one concern - how to migrate the installations from the internal plugin to an external one (as my understaning is that the internal implementation is still used), without losing any data. But there are probably more issues than this one

Also, is this something being actively worked on? If not, then I am willing to at least look into it. Although I am more of a C# hobbyist than a full-fledged developer, I believe I can get around in the code sufficiently well.

**Comment by Shadowghost** [comment]
The plugin living inside the server is more recent than the extracted one. The main blocker for extracting the core plugins (tmdb, musicbrainz, etc) from core was missing in-process restarting. That is implemented now, so in theory the plugins can be extracted (again) but we'd also need the possibility to install plugins on-demand in the setup wizard before this is a viable option.

**Reply by lyarenei** [reply]
> The main blocker for extracting the core plugins (tmdb, musicbrainz, etc) from core was missing in-process restarting. That is implemented now ...

Ah, okay, that's good to hear.

> we'd also need the possibility to install plugins on-demand in the setup wizard before this is a viable option.

Is this absolutely necessary though? I mean, I think having a barebone server without any plugins is fine too - at least in the first iteration. Then you can improve on it later by adding such option.

**Reply by nielsvanvelzen** [reply]
We decided that for the UX it is necessary to allow plugin installation during setup. with no plugins you would not have any metadata provider and Jellyfin will basically be useless.

The setup flow would be something like this:

1. create admin account
2. add first library
2.1. select folder
2.2. select metadata provider
3. finish setup
--server installs plugins and restarts--
4. welcome to jellyfin

**Reply by lyarenei** [reply]
@nielsvanvelzen Yeah, I was just suggesting that a bare Jellyfin server would be fine too. But when you put it that way, then I agree it'd feel a bit empty and perhaps a bit underwhelming as well.

Unfortunately, js/frontend is not really one of my strengths, so I am afraid I cannot help with that initiative. 

---
For now, I think the option I can go with is to make the changes in the plugin inside server. ~This means no 3rd party dependencies, right? I planned to propose using [this client](https://github.com/Zastai/MetaBrainz.MusicBrainz) in the plugin - which would clean it up quite nicely. But in this state, I think it's a bad idea to have such dependency in the server build. And I'm not sure if you would be fine with it even if the plugin was standalone already.~ I just looked at the source and it's already used, great! :)
